research-software-directory/RSD-as-a-service
# Maintainer roles

# Maintainer roles

Changes proposed in this pull request:

* when creating a software or project entry, you automatically become a maintainer
* maintainers can now edit software, projects and related things they maintain
* some automated tests were written using Postman that can be run with Newman

How to test:

* `docker-compose down --volumes`
* `docker-compose build database backend test`
* `docker-compose up database backend test`
* After a while, the test service should output a table which conaint a row `assertions` with 46 as `executed` value. All values in the column `failed` should have value zero.
* Row level security should behave as you expect: maintainers can edit exactly those rows belonging to software of project entries that they maintain. This was not all tested and should in the near future be tested both manually and with extra automated tests.

PR Checklist:


closes #16 
closes #17 
closes #22
